raft/rafttest/node.go

```diff
@@ -43,28 +50,36 @@ func startNode(id uint64, peers []raft.Peer, nt network) *node {
 				n.storage.Append(rd.Entries)
 				go func() {
 					for _, m := range rd.Messages {
-						nt.send(m)
+						n.nt.send(m)
 					}
 				}()
 				n.Advance()
 			case m := <-n.nt.recv():
 				n.Step(context.TODO(), m)
 			case <-n.stopc:
-				log.Printf("raft.%d: stop", id)
+				n.Stop()
+				log.Printf("raft.%d: stop", n.id)
+				n.Node = nil
 				return
 			}
 		}
 	}()
-	return n
 }
 
-func (n *node) stop() { close(n.stopc) }
-
-// restart restarts the node with the given delay.
-// All in memory state of node is reset to initialized state.
+// stop stops the node.
+// All in memory state of node is discarded.
 // All stable MUST be unchanged.
-func (n *node) restart(delay time.Duration) {
-	panic("unimplemented")
+func (n *node) stop() {
+	n.nt.disconnect(n.id)
+	close(n.stopc)
```

Comment:
i am aware of this. i was thinking we will not restart a stopped node immediately. 
i will fix it.